tensorflow_io/core/kernels/orc/orc_kernels.cc

```diff
@@ -0,0 +1,215 @@
+      for (uint32_t r = 0; r < batch->numElements; ++r) {
+        for (size_t column_index = 0; column_index < columns_.size();
+             column_index++) {
+          switch (dtypes_[column_index]) {
+            case DT_DOUBLE:
+              PROCESS_TYPE(orc::DoubleVectorBatch*, double, double);
+              break;
+            case DT_FLOAT:
+              PROCESS_TYPE(orc::DoubleVectorBatch*, double, float);
+              break;
+            case DT_INT16:
+              PROCESS_TYPE(orc::LongVectorBatch*, int64, int16);
+              break;
+            case DT_INT32:
+              PROCESS_TYPE(orc::LongVectorBatch*, int64, int32);
+              break;
+            case DT_INT64:
+              PROCESS_TYPE(orc::LongVectorBatch*, int64, int64);
+              break;
+            case DT_STRING: {
+              auto* string_col = dynamic_cast<orc::StringVectorBatch*>(
+                  fields->fields[column_index]);
+              char** buffer = string_col->data.data();
+              int64_t* lengths = string_col->length.data();
+              tensors_[column_index].flat<tstring>()(record_index) =
+                  std::string(buffer[r], lengths[r]);
+              break;
+            }
+            default:
+              return errors::InvalidArgument(
+                  "data type is not supported: ",
+                  DataTypeString(dtypes_[column_index]));
+          }
+        }
+        record_index++;
+      }
+    }
+
+    return Status::OK();
+  }
+
+  Status Read(const int64 start, const int64 stop, const string& component,
+              int64* record_read, Tensor* value, Tensor* label) override {
+    if (columns_index_.find(component) == columns_index_.end()) {
+      return errors::InvalidArgument("component ", component, " is invalid");
+    }
+    int64 column_index = columns_index_[component];
+
+    (*record_read) = 0;
+    if (start >= shapes_[column_index].dim_size(0)) {
+      return Status::OK();
+    }
+    const string& column = component;
+    int64 element_start = start < shapes_[column_index].dim_size(0)
+                              ? start
+                              : shapes_[column_index].dim_size(0);
+    int64 element_stop = stop < shapes_[column_index].dim_size(0)
+                             ? stop
+                             : shapes_[column_index].dim_size(0);
+    if (element_start > element_stop) {
+      return errors::InvalidArgument("dataset ", column,
+                                     " selection is out of boundary");
+    }
+    if (element_start == element_stop) {
+      return Status::OK();
+    }
+    for (int i = element_start; i < element_stop; i++) {
+      if (dtypes_[column_index] == DT_STRING) {
+        value->flat<tstring>().data()[i] =
+            tensors_[column_index].flat<tstring>().data()[i];
+        // LOG(INFO) << " column_index: " << column_index << " index: " << i <<
+        // " data: " << value->flat<tstring>().data()[i];
```

Comment:
Should we uncomment so that log will show up?

Reply:
this is primarily for debugging purpose, let me uncomment and use DEBUG